flight_clients/cli/client.py

```diff
@@ -0,0 +1,40 @@
+import logging
+import sys
+
+import requests
+
+if len(sys.argv) < 2:
+    logging.error("Please specify a host")
+    sys.exit(1)
+
+URL = 'http://{}:8000/commands/'.format(sys.argv[1])
+
+
+def gen_req(command, **kwargs):
+    resp = requests.post(URL, data={"command": command, "meta": kwargs})
+    if resp.status_code != requests.codes.ok:
+        logging.error(resp.text)
+
+
+while True:
```

Comment:
If you think it would be good, you could use the Command class which is used in user_command.py. The benefit is that it does type checking for each of the parameters of each command, and it prevents an ever growing list of elif statements.

Reply:
I do agree that in terms of style, `if elif elif elif` is not ideal, and I will think about a good way to solve that, but at the current moment, I am not convinced that your solution is any more elegant, it seems much less readable. For @christopher-o-toole's sake however, using the dictionary to solve this makes it instant lookup(O(1)), which is definitely... something... (maybe nitpicking)

Reply:
I agree that the way I did it is barely readable. The thing I dislike the most about the elif elif elif way of doing it is that there is lots of indexing by constants into the args variable, and because each command can have a different ordering of arguments, it makes things hard to understand. Maybe if we standardized the way a command is sent, it would help.
```suggestion
COMMAND_NAME = 0
COMMAND_PRIORITY = 1
CUSTOM_ARGS_START = 2
...
name = args[COMMAND_NAME]
priority = args[COMMAND_PRIORITY]
custom_args = args[CUSTOM_ARGS_START:]
if name == "move":
    process_movement_args(custom_args)
    controller.add_movement_task(*custom_args, priority)
... 

```